src/collection.jl

```diff
@@ -130,15 +225,167 @@ end
 
 
 """
-    images(df::DataFrame)
+    ccds(df::DataFrame)
 
-Generator for `ImageHDU`s of entries in data frame.
+Generator for `CCDData`s of entries in data frame.
+
+Iterates over `collection` using each `path` and `hdu` to load data using [`FITSIO.FITS`](http://juliaastro.github.io/FITSIO.jl/latest/api.html#FITSIO.FITS) into a [`CCDData`](@ref).
+
+# Examples
+```julia
+collection = fitscollection("~/data/tekdata")
+for hdu in ccds(collection)
+    @assert hdu isa CCDData
+end
+```
 """
-function images end
+function ccds end
 
-# generator for ImageHDU specified by data frame (i.e. path of file, hdu etc.)
-@resumable function images(df::DataFrame)
+# generator for CCDData specified by data frame (i.e. path of file, hdu etc.)
+@resumable function ccds(df::DataFrame)
```

Comment:
> Theoretically the input could just be a vector of named tuples, if they're structured correctly!

I think the user would have to dig in a bit for putting up a correct vector of named tuples, and if the user doesn't do it correctly, it may lead to errors. That's why I thought of explicitly writing `DataFrame` in the description to avoid this confusion.

Although, only writing `DataFrame` would not make sense, let's modify the docs, and keep it as:
`df` can be a vector of named tuples or `DataFrame` with some specific properties.

Also, I am not sure about the compatibility of `eachrow` with vector of named tuples.

Reply:
Actually, a user has to do extremely little to transfer between table types thanks to the Tables.jl interface. I'm away from my computer but I can easily take a dataframe from the collection function, turn it into a TypedTable from TypedTables.jl and there's no reason that shouldn't work with our function. So specializing in DataFrame means the users can't use anything else. My point is that we don't need the specialization AND it restricts our users, so why have it?

Reply:
Ya, that's a point. Let's drop it!